spec/unit/ssl/certificate_request_spec.rb

```diff
@@ -335,10 +335,51 @@
       expect(csr.verify(key.content)).to be_truthy
     end
 
-    it "should raise an error if neither SHA256 nor SHA1 are available" do
+    # Attempts to use SHA512 and SHA384 for signing certificates don't seem to work
```

Comment:
Will this be addressed as a separate ticket? Does it block workflows that set digest_algorithm to sha384 or sha512?

Reply:
Yes this should/will be addressed sometime but is not a high priority. It seems peculiar to when running it as spec test; works fine using a regular ruby script. Agents can get/apply catalog fine when using either of these as their default digest_algorithm (CSR do not seem to use configurable digest_algorithm). I will file a ticket. 
I have initiated a CI build to run all acceptance tests in case there are tests that not get exercised by spec tests; will update with its results.

Reply:
I have filed PUP-8110 for the above. 
I ran puppet acceptance tests thru CI while setting the digest_algorithm to sha384 and all but one pass (parser_functions/calling_all_functions.rb. Attached log. I had to update a couple of acceptances to use sha384 (instead of md5). I re-ran the tests using sha256 (which was already supported) to compare but it shows same failure in the above test. Attached log. Which leads me to think that being a test issue (would like to understand how to fix that test though).  
[puppet-CI-acceptance-runs-sha256-384.tar.gz](https://github.com/puppetlabs/puppet/files/1432266/puppet-CI-acceptance-runs-sha256-384.tar.gz)